src/m365/aad/commands/app/app-role-delete.ts

```diff
@@ -0,0 +1,265 @@
+import { Cli, Logger } from '../../../../cli';
+import {
+  CommandOption
+} from '../../../../Command';
+import GlobalOptions from '../../../../GlobalOptions';
+import request from '../../../../request';
+import GraphCommand from '../../../base/GraphCommand';
+import commands from '../../commands';
+import {Application, AppRole} from "@microsoft/microsoft-graph-types";
+import Utils from '../../../../Utils';
+
+interface CommandArgs {
+  options: Options;
+}
+
+interface Options extends GlobalOptions {
+  appId?: string;
+  appObjectId?: string;
+  appName?: string;
+  claim?: string;
+  name?: string;
+  id?:string;
+}
+
+class AadAppRoleDeleteCommand extends GraphCommand {
+
+  public get name(): string {
+    return commands.APP_ROLE_DELETE;
+  }
+
+  public get description(): string {
+    return 'Deletes role from the specified Azure AD app registration';
+  }
+
+  public getTelemetryProperties(args: CommandArgs): any {
+    const telemetryProps: any = super.getTelemetryProperties(args);
+    telemetryProps.appId = typeof args.options.appId !== 'undefined';
+    telemetryProps.appObjectId = typeof args.options.appObjectId !== 'undefined';
+    telemetryProps.appName = typeof args.options.appName !== 'undefined';
+    telemetryProps.claim = typeof args.options.claim !== 'undefined';
+    telemetryProps.name = typeof args.options.name !== 'undefined';
+    telemetryProps.id = typeof args.options.id !== 'undefined';
+    return telemetryProps;
+  }
+
+  public commandAction(logger: Logger, args: CommandArgs, cb: () => void): void {
+
+    const deleteAppRole: () => void = (): void => { 
+      this.processAppRoleDelete(logger,args).then(_ => cb(), (rawRes: any): void => this.handleRejectedODataJsonPromise(rawRes, logger, cb));
+    };
+
+    if (args.options.confirm) {
+      deleteAppRole();
+    }
+    else {
+      Cli.prompt({
+        type: 'confirm',
+        name: 'continue',
+        default: false,
+        message: `Are you sure you want to delete the App role ?`
+      }, (result: { continue: boolean }): void => {
+        if (!result.continue) {
+          cb();
+        }
+        else {
+          deleteAppRole();
+        }
+      });
+    }
+  }
+
+  private processAppRoleDelete(logger: Logger, args: CommandArgs):Promise<void>{
+    return this
+      .getAppObjectId(args, logger)
+      .then((appId: string) => this.getAADApp(appId, logger))
+      .then((aadApp: Application): Promise<any> => {
```

Comment:
Since `processAppRoleDelete` has a return type of `Promise<void>`, this last promise should also have a return type of `Promise<void>` rather than `Promise<any>`

Reply:
Got it. Fixed